analyzers/src/SonarAnalyzer.Common/Rules/AspNet/RouteTemplateShouldNotStartWithSlashBase.cs

```diff
@@ -75,12 +75,10 @@ private void ReportIssues(SonarSymbolReportingContext context, INamedTypeSymbol
             ? MessageOnlyActions
             : MessageActionsAndController;
 
-        var secondaryLocations = actions.SelectMany(x => x.RouteParameters.Keys);
-        foreach (var classDeclaration in controllerSymbol.DeclaringSyntaxReferences.Select(x => x.GetSyntax()))
+        var secondaryLocations = actions.SelectMany(x => x.RouteParameters.Keys).Select(x => x.ToSecondary());
```

Comment:
This is about the 10th time in this PR that I see a `Select( x=> x.ToSecondary())`. Consider adding `ToSecondary()` extension methods for `IEnumerable<SyntaxNode|SyntaxToken|Location>` as well.